src/common/download.h

```diff
@@ -32,5 +32,11 @@
 
 namespace tools
 {
-  bool download(const std::string &path, const std::string &url);
+  typedef void *download_async_handle;
```

Comment:
Used to be the caller would call an unref function, but now it doesn't: it just has to set its pointer to NULL in the result callback (which is locked by the internal lock there, so is safe). The object is now only deleted by the download code itself.